src/chaos/controller.go

```diff
@@ -77,27 +77,46 @@ func (cc *ChaosController) deleteNode(namespace string, podName string) error {
 	return cc.DynamicClient.Resource(podGVR).Namespace(namespace).Delete(context.TODO(), podName, v1.DeleteOptions{})
 }
 
-// configConfigPath    - path to k8s cluster config
-// killNodes 	- to kill or not to kill affected nodes after each run
-func RunChaosExperiment(namespace string, chaosConfigPath string) error {
+func RunChaosExperiment(namespace string, manifestObj *unstructured.Unstructured) error {
 	cc := NewController(namespace)
 
-	// Work with experiment yaml file
-	experimentFile, err := os.ReadFile(chaosConfigPath)
+	expType, _, err := unstructured.NestedString(manifestObj.Object, "kind")
+	if err != nil {
+		return err
+	}
+	gvr := schema.GroupVersionResource{
+		Group:    "chaos-mesh.org",         // ChaosMesh group
+		Version:  "v1alpha1",               // Version of the ChaosMesh resource
+		Resource: strings.ToLower(expType), // Resource is "podchaos"
+	}
+
+	// Apply the resource (equivalent to `kubectl apply -f`)
+	_, err = cc.DynamicClient.Resource(gvr).Namespace(cc.Namespace).Create(context.TODO(), manifestObj, v1.CreateOptions{})
+	if err != nil {
+		log.Fatalf("Failed to apply ChaosMesh resource: %v", err)
+	}
```

Comment:
You don't need these lines in two functions. You can run RunChaosExperiment function from RunChaosExperimentFile.